### PR TITLE
Search README issues

### LINKED ISSUE #13781: sdk/search/README.md lists track 2 as "preview", fully documents track 1
I'm not sure how much we rely on README.md in the service client roots, but I see that sdk/search/README.md currently reads,

````markdown
Functionality is exposed through several client libraries:

- [Azure.Search.Documents](./Azure.Search.Documents/) is a preview experience
  built on top of [Azure.Core](../core/Azure.Core/README.md) and the
  [Azure SDK Design Guidelines for .NET](https://azure.github.io/azure-sdk/dotnet_introduction.html).

- [Microsoft.Azure.Search](./Microsoft.Azure.Search/) is the officially
supported .NET client library that makes it easy to develop applications using
Azure Cognitive Search.  The `Microsoft.Azure.Search` package includes the
following client libraries:
  - [Microsoft.Azure.Search.Data](./Microsoft.Azure.Search.Data/) enables
````

Note track 2 reads "a preview experience". We should at least update that, if not downplay track 1.

### PR DESCRIPTION
**Pull request by tg-msft**
Fixes #13781 and superficially addresses #13502 (which will be left open to fix properly next release)

### CODE REVIEW

File: sdk/search/README.md [hunk 1]
```diff
@@ -15,14 +15,15 @@ persisted index that you create, own, and store in your service.
 
 Functionality is exposed through several client libraries:
 
-- [Azure.Search.Documents](./Azure.Search.Documents/) is a preview experience
-  built on top of [Azure.Core](../core/Azure.Core/README.md) and the
+- [Azure.Search.Documents](./Azure.Search.Documents/) is the latest .NET client
+  library for building applications with Azure Cognitive Search.  It is built on
+  top of [Azure.Core](../core/Azure.Core/README.md) and the
```
Comment (heaths) [inline sdk/search/README.md:20]: This should be a fully-qualified path to Azure.Core's README.md (master branch), given the new recommendations.

File: sdk/search/Azure.Search.Documents/README.md [hunk 2]
```diff
@@ -249,8 +249,8 @@ SearchResults<SearchDocument> response = client.Search<SearchDocument>("luxury")
 foreach (SearchResult<SearchDocument> result in response.GetResults())
 {
     SearchDocument doc = result.Document;
-    string id = (string)doc["hotelId"];
-    string name = (string)doc["hotelName"];
+    string id = (string)doc["HotelId"];
```
Comment (heaths) [inline sdk/search/Azure.Search.Documents/README.md:252]: But all the other samples above use camelCase field names? I thought the plan was to document switching over to the public read-only Hotels sample, which should apply for the whole of the README samples, code and markdown.

Comment (tg-msft) [reply sdk/search/Azure.Search.Documents/README.md:252]: That is still the plan.  It's just not happening in time for this release so I'm fixing up the README manually in the short term.  The other samples with field names are all creating an index which might not perfectly match the same index you'd get from the Portal, but isn't going to lead to any bugs.

**Review by heaths — APPROVED** [review]
After talking, the other samples need to have the casing fixed as needed, like the "Using C# Types" and "SearchOptions" sections. Signing off to unblock.

**Pull request merged**